packages/core/saved-objects/core-saved-objects-api-server/src/apis/get.ts

```diff
@@ -16,4 +16,5 @@ import { SavedObjectsBaseOptions } from './base';
 export interface SavedObjectsGetOptions extends SavedObjectsBaseOptions {
   /** {@link SavedObjectsRawDocParseOptions.migrationVersionCompatibility} */
   migrationVersionCompatibility?: 'compatible' | 'raw';
+  versionModelMatch?: 'strict';
```

Comment:
Ihmo the optional parameter with a single allowed value is a bad pattern. I would do the same than what we did for `migrationVersionCompatibility` and allow to specify both values (while still defaulting to the current behavior)

something like

```
versionModelMatch: 'strict' | 'lax'
```

Also, (probably nitpicking, but) I'm not sure `versionModelMatch` is an appropriate naming, given it would only block conversions in one way (we still allow to upgrade documents regardless of this option's value).

Maybe something that makes it more explicit that this is only going to be used for "downward" conversions.

Maybe something like:

```
downwardConversion: 'allow' | 'forbid'
```

WDYT?

Reply:
Your advice is awesome, thank you! Naming is hard and I didn't want to get hung up about it. 

I'll make the changes.

Reply:
When would 'lax' be used? The condition won't be used anywhere sine I'm using a ternary to decide when to throw from docMigrator.